api/v1alpha1/clusterinstance_types.go

```diff
@@ -172,6 +172,11 @@ type NodeSpec struct {
 	// +optional
 	IronicInspect IronicInspect `json:"ironicInspect,omitempty"`
 
+	// PreprovisioningNetworkDataName is the name of the Secret in the 	// local namespace containing network
+	// configuration (e.g content of network_data.json) which is passed to the preprovisioning image, and to the
+	// Config Drive if not overridden by specifying NetworkData.
+	PreprovisioningNetworkDataName string `json:"preprovisioningNetworkDataName,omitempty"`
```

Comment:
Why expose this name as API?

Did we do that previously for any other secret or configmap?

Reply:
Removed this as per our conversation. tl;dr: the NMState secret will be used in stead of the ConfigMap.